### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by neurolabusc**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+## Node.JS Command Line Demo
+
+The module `streamlineIO.mjs` provides JavaScript reading for streamlines in the TCX, TRK, TRX, and VTK formats. The included file `bench.mjs` demonstrates these capabilities by loading a streamline 11 times and reporting details regarding file size, loading time, and streamline properties.
+
+To run the demo:
+
+```
+$ git clone https://github.com/tee-ar-ex/trx-javascript
+$ cd trx-javascript
+$ npm install gl-matrix fflate fzstd
```
Comment (frheault) [inline README.md:31]: Is it possible to write what is the correct version of Node?
I used `sudo apt` and the version was so old the code kept crashing on the first line.
When I installed it directly from a `.tar.gz` from their website it worked immediately. 

Something like?
`Install the appropriate version of the most recent [NodeJS](https://nodejs.org/en/download/).`

**Comment by neurolabusc** [comment]
@frheault one nice thing about Debian releases is they freeze software to make user experience consistent. This can also be frustrating when the user wants new features. I have updated the readme as you suggested. Note that node is only required for using the javascript from the command line. The code is compatible with all modern web browsers, as demonstrated by the [NiiVue live demo](https://niivue.github.io/niivue/features/tracts.html).

**Comment by neurolabusc** [comment]
@frheault I do think the revised PR addresses your concerns. If you agree, can you accept and apply this pull request.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+## About
+
+This is a minimal JavaScript reader for the [TRX tractography file format](https://github.com/tee-ar-ex/trx-spec/blob/master/specifications.md). Previously, most tractography tools used their own proprietary [format](https://www.nitrc.org/plugins/mwiki/index.php/surfice:MainPage#Supported_Formats) such as BFLOAT, niml.tract, PDB, [TCK](https://mrtrix.readthedocs.io/en/latest/getting_started/image_data.html#tracks-file-format-tck), [TRK](http://trackvis.org/docs/?subsect=fileformat) and VTK. The TRX format was developed from community [discussions](https://github.com/nipy/nibabel/issues/942) to address the [limitations of the existing formats](https://docs.google.com/document/d/1GOOlG42rB7dlJizu2RfaF5XNj_pIaVl_6rtBSUhsgbE/edit#heading=h.c6igqydj1hrf) and the needs of the users. This reader supports all the features of TRX including data-per-vertex, data-per-streamline and groups.
```
Comment (arokem) [inline README.md]: Could add a link to a GitHub repo that describes the PDB format: 
```suggestion
This is a minimal JavaScript reader for the [TRX tractography file format](https://github.com/tee-ar-ex/trx-spec/blob/master/specifications.md). Previously, most tractography tools used their own proprietary [format](https://www.nitrc.org/plugins/mwiki/index.php/surfice:MainPage#Supported_Formats) such as BFLOAT, niml.tract, [PDB](https://github.com/vistalab/pdb_files), [TCK](https://mrtrix.readthedocs.io/en/latest/getting_started/image_data.html#tracks-file-format-tck), [TRK](http://trackvis.org/docs/?subsect=fileformat) and VTK. The TRX format was developed from community [discussions](https://github.com/nipy/nibabel/issues/942) to address the [limitations of the existing formats](https://docs.google.com/document/d/1GOOlG42rB7dlJizu2RfaF5XNj_pIaVl_6rtBSUhsgbE/edit#heading=h.c6igqydj1hrf) and the needs of the users. This reader supports all the features of TRX including data-per-vertex, data-per-streamline and groups.
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+## Benchmark
+
+The included JavaScript `bench` provides a method to evaluate performance. This benchmark is likely specific to JavaScript and so caution should be excercised in evaluating relative performance. The script will report the time to load a TRK, TCK, VTK or TRX file 10 times (it loads the tracts 11 times, and ignores the first run).
+
+The graph below shows the time load the [left inferior fronto-occipital fasciculus (IFOF) ](https://brain.labsolver.org/hcp_trk_atlas.html) with 30856 streamlines and 11437105 vertices. The benchmark was run on a passively-cooled 15w M1-based MacBook Air. The ideal format would be both fast to load (to the left on the horizontal axis) and have a small file size (toward the bottom in the right axis). However, compression typically trades load time for file size. Here all data is loaded from a local solid state drive, whereas smaller files would benefit if data was loaded using a slow internet connection. The following file formats are illustrated (except where noted, both positions and indeices are stored with 32-bit precision):
```
Comment (arokem) [inline README.md]:
```suggestion
The graph below shows the time load the [left inferior fronto-occipital fasciculus (IFOF) ](https://brain.labsolver.org/hcp_trk_atlas.html) with 30856 streamlines and 11437105 vertices. The benchmark was run on a passively-cooled 15w M1-based MacBook Air. The ideal format would be both fast to load (to the left on the horizontal axis) and have a small file size (toward the bottom in the right axis). However, compression typically trades load time for file size. Here all data is loaded from a local solid state drive, whereas smaller files would benefit if data was loaded using a slow internet connection. The following file formats are illustrated (except where noted, both positions and indices are stored with 32-bit precision):
```

**Review by arokem — COMMENTED** [review]
A couple of small documentation suggestions/fixes from my end. How was the example file generated? Would it make sense to also put it on figshare? See also discussions surrounding https://github.com/tee-ar-ex/trx-python/pull/32

**Comment by neurolabusc** [comment]
@arokem I am happy with all your changes. Feel free to apply them and submit this initial PR. 

I do plan to make testing datasets available, but was waiting for a decision on the [final format](https://github.com/tee-ar-ex/trx-python/pull/33). My reader will work fine with or without the proposed modification, but I want the exemplar data to reflect the finalized format.

File: README.md (hunk 1 above)
Comment (arokem) [reply README.md:31]: @neurolabusc : do you want to address this question in this PR, or should we merge and add this as an issue for future work?

**Pull request merged**